streams/src/main/java/org/apache/kafka/streams/processor/internals/StateDirectory.java

```diff
@@ -102,6 +107,13 @@ public StateDirectory(final StreamsConfig config, final Time time, final boolean
             log.warn("Using /tmp directory in the state.dir property can cause failures with writing the checkpoint file" +
                 " due to the fact that this directory can be cleared by the OS");
         }
+        final Path path = Paths.get(baseDir.getPath());
```

Comment:
Is changing the `baseDir` sufficient?

Reply:
It should be, the `stateDir` is inside of the `baseDir` so the permissions should apply there as well. I guess if we wanted to be _more_ selective we could apply permissions to `stateDir` and not `baseDir` but from what I see it doesn't make a big difference

Reply:
I'm not sure off the top of my head I just wanted to verify, but its probably best to be overly specific. LGTM